chia/_tests/core/full_node/test_full_node.py

```diff
@@ -2299,9 +2300,18 @@ async def validate_coin_set(coin_store: CoinStore, blocks: list[FullBlock]) -> N
         prev_hash = block.header_hash
         rewards = block.get_included_reward_coins()
         records = {rec.coin.name(): rec for rec in await coin_store.get_coins_added_at_height(block.height)}
+
+        # validate reward coins
+        for reward in rewards:
+            rec = records.pop(reward.name())
```

Comment:
what was the purpose of moving this here ?

Reply:
I remove the records from the database that has been accounted for from the actual block in order to ensure the list is empty at the end. Otherwise I couldn't detect *additional* coin entries in the database, that don't exist in the block.